packages/framework/src/Component/EntityExtension/EntityExtensionSubscriber.php

```diff
@@ -0,0 +1,306 @@
+     */
+    protected function isOneOfRegisteredEntities(string $entityClass): bool
+    {
+        if ($this->allRegisteredEntitiesCache === null) {
+            $metadataDriver = $this->configuration->getMetadataDriverImpl();
+
+            if ($metadataDriver === null) {
+                throw new LogicException('Metadata driver cannot be null');
+            }
+
+            $this->allRegisteredEntitiesCache = $metadataDriver->getAllClassNames();
+        }
+
+        return in_array($entityClass, $this->allRegisteredEntitiesCache, true);
+    }
+
+    /**
+     * @param class-string $entityClass
+     * @return \Doctrine\ORM\Mapping\ClassMetadataInfo
+     */
+    protected function getClassMetadataForEntity(string $entityClass): ClassMetadataInfo
+    {
+        $classMetadata = new ClassMetadataInfo(
+            $entityClass,
+            $this->configuration->getNamingStrategy()
+        );
+
+        $metadataDriver = $this->configuration->getMetadataDriverImpl();
+
+        if ($metadataDriver === null) {
+            throw new LogicException('Metadata driver cannot be null');
+        }
+
+        $metadataDriver->loadMetadataForClass($entityClass, $classMetadata);
+
+        return $classMetadata;
+    }
+
+    /**
+     * @param \Doctrine\ORM\Mapping\ClassMetadataInfo $classMetadata
+     */
+    protected function updateAssociationMappingsToMappedSuperclasses(ClassMetadataInfo $classMetadata): void
+    {
+        foreach ($classMetadata->getAssociationMappings() as $name => $mapping) {
+            if (!isset($mapping['targetEntity']) || !$this->isParentEntity($mapping['targetEntity'])) {
+                continue;
+            }
+
+            $overridingClass = $this->getOverridingClass($mapping['targetEntity']);
+
+            $mapping['targetEntity'] = $overridingClass;
+            $classMetadata->associationMappings[$name] = $mapping;
+        }
+    }
+
+    /**
+     * @param class-string $className
+     * @return class-string|null
+     */
+    protected function getOverridingClass(string $className): ?string
+    {
+        if (array_key_exists($className, $this->entityExtensionMap)) {
+            return $this->entityExtensionMap[$className];
+        }
+
+        foreach ($this->parentClassesByClass as $extendedClass => $parentClassesByClass) {
+            foreach ($parentClassesByClass as $class) {
+                if ($className === $class) {
+                    return $extendedClass;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param string $classNameCandidate
+     * @param class-string $parentClass
+     * @return class-string
+     */
+    protected function ensureAbsoluteClassName(string $classNameCandidate, string $parentClass): string
+    {
+        if (class_exists($classNameCandidate) || str_contains($classNameCandidate, '\\')) {
+            return $classNameCandidate;
+        }
+
+        $reflClass = new ReflectionClass($parentClass);
```

Comment:
nitpick - we should not use shortcuts, I would rather see `$reflectionClass` variable name here :innocent:

Reply:
true indeed :)